src/inmanta/env.py

```diff
@@ -696,6 +696,12 @@ def init_env(self) -> None:
                 raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path}")
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
         elif not os.path.exists(self._path_pth_file):
+            if not os.path.exists(self.site_packages_dir):
+                # the venv hosts a different python version than the running process
+                raise VenvCreationFailedError(
+                    msg=f"Unable to create new virtualenv at {self.env_path} due to python version mismatch"
```

Comment:
```suggestion
                    msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch"
```
It doesn't necessarily get created. If it already exists (which is the case here), we just activate it.